### ISSUE TITLE
2.1 doesnt respect print logs to console parameter

### ISSUE THREAD

**Issue by nate-strauser** [superstringsoftware/observatory#19]
if i init like
`var log = TLog.getLogger(TLog.LOGLEVEL_MAX, false, true, true);`

i get this in the console
`LOG: [26/2/2013][20:38:49.6][CLIENT][Observatory][VERBOSE][] Creating logger with level DEBUG, print to console: true, log user: true`

creates problems for IE browsers for me

v0.2.0 works well with
`var log = TLog.getLogger(TLog.LOGLEVEL_MAX, false, true);`

**Comment by jhoxray** [comment]
Hi Nate

It's a feature, not a bug - now you are supposed to do getLogger() and then log.setOptions(...) explicitly (same signature).

The issue was that since TLog is currently a singleton if somebody calls TLog.getLogger() for instance from a package they will mess up the settings you have in your app, so now getLogger() does not change the settings - and if it's a first call, sets them to default where print to console is true.

Bottom line: call setOptions(..) right after getLogger for now, in the future we'll change it from Singleton to a regular class so that packages can create their own instances and not interfere with the main app. It also has other uses and seems a better design decision now in real-world apps.

That said, changing default to not output to console is a small fix that makes sense, will add that.

**Comment by nate-strauser** [comment]
hmm - ok - i updated to the new pattern - setting options works well after the init - however i'm not finding a way to suppress the very first log message from hitting the browser console

![Screen Shot 2013-03-27 at 10 24 44 AM](https://f.cloud.github.com/assets/389646/308543/5e2decb8-96ea-11e2-8df6-7d143a88c7a1.png)

![Screen Shot 2013-03-27 at 10 24 51 AM](https://f.cloud.github.com/assets/389646/308544/6345b794-96ea-11e2-8f82-13d94759406f.png)

that single message output to the console is troublesome - i havent checked in this version, but in 1.8 any output to console halted javascript execution on certain IE versions - maybe default to console output off if client and on if server?

**Comment by nate-strauser** [comment]
did a little quick testing - doesnt seem to cause any issues in IE8/9 any more - so this is just a minor annoyance

**Comment by nate-strauser** [comment]
have to correct myself - did a little more testing today - the initial print to console does indeed kill IE9 for me

to replicate with an app that inits client side
open ie9, make sure dev toolbar is closed, close ie, open again (do not open dev tools), navigate to app - first console statement stops all javascript execution which results in just a white/blank page

if dev tools are opened in the same browser session before hitting the app, it works fine - see http://stackoverflow.com/questions/5472938/does-ie9-support-console-log-and-is-it-a-real-function

i tried various console.log hacks/fixes for ie right before initing the logger, but they arent resolving the problem for me

not sure if you'd want to have console printing off by default, but that seems like it would take care of this

**Comment by jhoxray** [comment]
Thanks Nate, will do that, if somebody needs to print to console they can always set this option.
